Notify/Notify/Notify/Helpers/Converter.cs

```diff
@@ -19,6 +20,107 @@ public object ConvertBack(object value, Type targetType, object parameter, Cultu
+        }
+
+        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            return null;
+        }
+    }
+
+    public class BoolToEditIconConverter : IValueConverter
+    {
+        // Similar logic here, return "edit_button.png" or "edit_button_disabled.png"
+        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            return (bool)value ? "edit_button" : "edit_button_disabled";
+
+        }
+
+        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            return null;
+        }
+    }
+
+    public class BoolToRenewIconConverter : IValueConverter
+    {
+        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            return (bool)value ? "renew_button" : "renew_button_disabled";
+        }
+
+        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            return null;
+        }
+    }
+    
+    public class TypeToColorConverter : IValueConverter
+    {
+        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            Color notificationColor = Color.Default;
+            
+            if (value is NotificationType type)
+            {
+                switch (type)
+                {
+                    case NotificationType.Location:
+                        notificationColor = Constants.LOCATION_NOTIFICATION_COLOR;
+                        break;
+                    case NotificationType.Dynamic:
+                        notificationColor = Constants.DYNAMIC_NOTIFICATION_COLOR;
+                        break;
+                    case NotificationType.Time:
+                        notificationColor = Constants.TIME_NOTIFICAATION_COLOR;
+                        break;
+                }
+            }
+
+            return notificationColor;
+        }
+
+        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            return null;
+        }
+    }
+    
+    public class TypeToIconConverter : IValueConverter
+    {
+        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            if (value is NotificationType type)
+            {
+                switch (type)
+                {
+                    case NotificationType.Location:
+                        return ImageSource.FromFile("location_colored_icon"); 
+                    case NotificationType.Dynamic:
+                        return ImageSource.FromFile("dynamic_location_colored_icon");
+                    case NotificationType.Time:
+                        return ImageSource.FromFile("time_colored_icon");
+                    default:
+                        return null; 
```

Comment:
I think that an exception should be thrown here, for passing invalid argument.

Reply:
RIght